File: Python/to_weird_case.py
def to_weird_case(string):
    word_list = string.split(' ')
    weird_list = []
    for each_word in word_list.copy():
        weird_word = ""
        for index, each_letter in enumerate(each_word):
            if index % 2 == 0: 
                weird_word += each_letter.upper()
            else:
                weird_word += each_letter
        print(weird_word)
        weird_list.append(weird_word)
    return ' '.join(weird_list)
    
print(to_weird_case('This is a test'))
# ThIs Is A TeSt

File: Javascript/toWeirdCase.js
function toWeirdCase(string){
  splitString = string.split(' ')
  weirdList = [];
  weirdString = ''
  for (let i = 0; i < splitString.length; i++){
    let evenCap = ''
    for (let j = 0; j < splitString[i].length; j++){
       j % 2 == 0 ? evenCap += splitString[i][j].toUpperCase() : evenCap += splitString[i][j];
    }
    weirdList.push(evenCap)
  }
  weirdString = weirdList.join(' ')
  return weirdString
}
console.log(toWeirdCase('This is a test'))
//ThIs Is A TeSt